src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonUcImplSecurityConstraintCheck.java

```diff
@@ -0,0 +1,159 @@
+
+    ClassTree tree = getClassTree(context);
+    TypeTree ucInterface = getUcInterface(tree);
+
+    if (ucInterface == null) {
+      return;
+    }
+
+    List<MethodTree> methodsOfTree = getMethodsOfTree(tree);
+    for (MethodTree method : methodsOfTree) {
+      if (!isMethodProperlyAnnotated(method)) {
+        context.reportIssue(this, method,
+            "This method is not properly annotated. "
+                + "Please use one of the following annotations on Use-Case implementation methods: "
+                + REQUIRED_ANNOTATIONS.toString());
+      }
+    }
+  }
+
+  /**
+   * @param context Context of analysis containing the parsed tree.
+   * @return Class tree.
+   */
+  protected ClassTree getClassTree(JavaFileScannerContext context) {
+
+    CompilationUnitTree parsedTree = context.getTree();
+    List<Tree> types = parsedTree.types();
+
+    for (Tree tree : types) {
+      if (tree instanceof ClassTree) {
+        return (ClassTree) tree;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Returns the Uc interface implemented by the checked class, if there is one.
+   *
+   * @param tree Tree currently being investigated.
+   * @return TypeTree instance of the Uc interface.
+   */
+  protected TypeTree getUcInterface(ClassTree tree) {
+
+    List<TypeTree> interfaces = tree.superInterfaces();
+    String className = tree.simpleName().name();
+    Pattern interfaceRegEx = Pattern.compile(className.replaceAll("Impl", ""));
+
+    if (interfaces.isEmpty()) {
+      return null;
+    }
+
+    for (TypeTree interfaceTree : interfaces) {
+      if (interfaceRegEx.matcher(interfaceTree.toString()).matches()) {
+        return interfaceTree;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Returns all methods of the given tree.
+   *
+   * @param tree Tree currently being investigated.
+   * @return List of MethodTree.
+   */
+  protected List<MethodTree> getMethodsOfTree(ClassTree tree) {
+
+    List<Tree> membersOfTree = tree.members();
+    List<MethodTree> methodsOfTree = new ArrayList<>();
+
+    for (Tree member : membersOfTree) {
+      if (member.is(Tree.Kind.METHOD)) {
+        methodsOfTree.add((MethodTree) member);
+      }
+    }
+
+    return methodsOfTree;
+  }
+
+  private boolean isMethodProperlyAnnotated(MethodTree method) {
+
+    List<AnnotationTree> annotationsOfMethod = method.modifiers().annotations();
+    boolean hasOverrideAnnotation = false;
+    boolean hasRequiredAnnotation = false;
+
+    for (AnnotationTree annotation : annotationsOfMethod) {
+      if (annotation.annotationType().toString().equals("Override")) {
+        hasOverrideAnnotation = true;
+      }
+
+      if (REQUIRED_ANNOTATIONS.contains(annotation.annotationType().toString())) {
+        hasRequiredAnnotation = true;
+      }
+    }
+
+    if ((!hasOverrideAnnotation) || (hasOverrideAnnotation && hasRequiredAnnotation)) {
+      return true;
+    } else {
+      return false;
+    }
+
+  }
```

Comment:
Nice and clean code. I like the style that you create smaller private methods with speaking name.
What I forgot to specify: we should also check that the method is `public` as the `Override` annotation could also be present for a `protected` method that can not come from the use-case interface. In such case we would create a false positive.